powerranger/powerranger.py

```diff
@@ -1,10 +1,136 @@
-import sys
+import curses
+import itertools
+import os
+from pathlib import Path
+from typing import Dict, Optional, Union
 
+DEFAULT_STARTUP_DIR = os.curdir
+SORT_FOLDERS_ON_TOP = True
 
-def main():
+
+class Colors:
+    """Wrapper around curses colors for Items."""
+    _pairs: Dict[str, int] = {}
+
+    @classmethod
+    def blue_on_black(cls):
+        """Initialized blue-on-black color pair."""
+        return cls._setup_pair("blue_black", curses.COLOR_BLUE, curses.COLOR_BLACK)
+
+    @classmethod
+    def yellow_on_black(cls):
+        """Initialized yellow-on-black color pair."""
+        return cls._setup_pair("yellow_black", curses.COLOR_YELLOW, curses.COLOR_BLACK)
+
+    @classmethod
+    def default(cls):
+        """Initialized white-on-black color pair."""
+        return cls._setup_pair("default", curses.COLOR_WHITE, curses.COLOR_BLACK)
+
+    @classmethod
+    def _setup_pair(cls, pair_name, foreground, background):
+        """Lazily configure and remember an ncurses color pair."""
+        if cls._pairs.get(pair_name) is None:
+            if not cls._pairs:
+                cls._pairs[pair_name] = 1
+            else:
+                cls._pairs[pair_name] = max(cls._pairs.values()) + 1
+
+            curses.init_pair(cls._pairs[pair_name], foreground, background)
+
+        return curses.color_pair(cls._pairs[pair_name])
+
+
+class Item:
+    """An item inside of a Directory."""
+    def __init__(self, path: Union[Path, str]):
+        self._path = Path(path)
+
+    @property
+    def name(self) -> str:
+        """The name of the item, not including parents."""
+        return self._path.name
+
+    @property
+    def color(self):
+        """An initialized ncurses color pair associated with the type of file
+        for this Item.
+        """
+        if self._path.is_dir():
+            return Colors.blue_on_black() | curses.A_BOLD
+
+        return Colors.default()
+
+
+class Directory:
+    """A list of items inside of a directory."""
```

Comment:
I'm not sure this docstring is accurate.

Reply:
The functionality of the class is likely to change very much more in the next iteration as well. I see no point in updating it until at least some functionality is petrified.